source/slang/slang-check-shader.cpp

```diff
@@ -913,19 +937,19 @@ namespace Slang
                     allComponentTypes.add(entryPoint);
                 }
             }
-        }
 
-        // Also consider entry points that were introduced via adding
-        // a library reference...
-        //
-        for( auto extraEntryPoint : compileRequest->m_extraEntryPoints )
-        {
-            auto entryPoint = EntryPoint::createDummyForDeserialize(
-                linkage,
-                extraEntryPoint.name,
-                extraEntryPoint.profile,
-                extraEntryPoint.mangledName);
-            allComponentTypes.add(entryPoint);
+            // Also consider entry points that were introduced via adding
+            // a library reference...
+            //
+            for (auto extraEntryPoint : compileRequest->m_extraEntryPoints)
```

Comment:
Moving this loop under the `if` changes the semantics in the case where *no* explicit entry points were added, because it makes it so that these "extra" entry points only get added if there were explicit entry point already (which should almost never be what you want).

Seeing this change here makes me wonder whether the whole change here isn't unnecessary.

If the original code that is invoking the Slang API here just didn't specify *any* explicit entry points, then it would auto-magically get these "extra" entry points from the linked libraries added in, and it could request kernel code from them just fine. This change seems to make it so that you can *only* use the "extra" entry points by specifying at least one explicit entry point, and that means you'd end up with duplication in `allComponentTypes` since the same entry point would be added both as an explicit entry point and as an "extra" one.